### PR TITLE
Feature: Add-test-framework for YAML

### LINKED ISSUE #120: ❓👂 Question/Feedback - Colorblind people cannot see the anchor icon, try different color combination in the dark theme
## Question/Feedback

Colorblind people cannot easily see the anchor icon, try different color combination in the dark theme.

## Possible Answers/Solutions?

Change cell background color or change icon color.

image

### PR DESCRIPTION
**Pull request by oZakari**
# Overview/Summary

Adding python test framework for YAML recommendation files and some other changes relevant to this framework such as aligning recommendations where applicable.

See test results for good yaml: https://github.com/oZakari/Azure-Proactive-Resiliency-Library-v2/actions/runs/9084684943
See test results for bad yaml: https://github.com/oZakari/Azure-Proactive-Resiliency-Library-v2/actions/runs/9084666973/job/24966144799

## Related Issues/Work Items

AB#34876
Fixes #120 

## This PR fixes/adds/changes/removes

1. Adds python script and associated schemas
2. Updates to recommendations to align to schema files
3. Save all files with UTF-8 encoding
4. Adds aprl-sap team as codeowners
5. Removed duplicate service health alert recommendation from WAF section

### Breaking Changes

1. None

## As part of this pull request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Review by ejhenry — APPROVED** [review]

**Review by ehaslett — APPROVED** [review]
LGTM

File: azure-waf/deploy/recommendations.yaml [hunk 1]
```diff
@@ -50,13 +50,13 @@
...
-  recommendationControl: Automation
+  recommendationControl: Other Best Practices
   recommendationImpact: Medium
   recommendationResourceType: WellArchitected/Deploy
   recommendationMetadataState: Active
   longDescription: |
     Another aspect of reliability is the deployment method used to update or roll back your workload. You should ensure that the deployment method used is aligned to your reliability strategy and supports the availability and performance requirements of your workload. For example, you may choose to use blue-green deployments, canary deployments, or rolling deployments depending on the level of risk and impact you want to mitigate. You should also consider the trade-offs between speed and safety when deploying changes, and use appropriate testing and validation methods to ensure the quality of your code and configuration. Additionally, you should have a clear and documented process for handling deployment failures and communicate any changes or issues to your stakeholders and customers.
-  potentialBenefits: Ensure deployment strategy satisfies reliability requirements
+  potentialBenefits: Ensure deployment strategy adheres reliability requirements
```
Comment (oZakari) [inline azure-waf/deploy/recommendations.yaml:59]:
```suggestion
  potentialBenefits: Align deployment strategy to reliability requirements
```

**Review by rodrigosantosms — APPROVED** [review]
Approved

File: azure-specialized-workloads/sap/recommendations.yaml [hunk 2]
```diff
@@ -372,15 +372,15 @@
     - name: ASCS-LB - Central Server Instance
       url: "https://docs.microsoft.com/en-us/azure/advisor/advisor-reference-reliability-recommendations"
 
-- description: DBHANA-Pacemaker (Database Instance) Ensure the Pacemaker cluster has been setup for SAP HANA DB high availability
+- description: Ensure the Pacemaker cluster has been setup for high availability
```
Comment (PmeshramPM) [inline azure-specialized-workloads/sap/recommendations.yaml:375]: This should say - "Ensure the Pacemaker cluster has been setup for SAP HANA DB high availability

File: azure-specialized-workloads/sap/recommendations.yaml [hunk 3]
```diff
@@ -349,12 +349,12 @@
     - name: ASCS-Pacemaker - Central Server Instance
       url: "https://docs.microsoft.com/en-us/azure/advisor/advisor-reference-reliability-recommendations"
 
-- description: ASCS-LB (Central Server Instance) Ensure the load balancer is configured correctly for SAP ASCS High availability
+- description: Ensure the central server load balancer is configured correctly for SAP ASCS High availability
```
Comment (PmeshramPM) [inline azure-specialized-workloads/sap/recommendations.yaml:352]: This should say - "Ensure the  load balancer is configured correctly for SAP ASCS High availability"

**Comment by oZakari** [comment]
Thanks @PmeshramPM, I have added your suggested changes. Please approve when you get a chance.

**Review by swathibhat1 — APPROVED** [review]
Reviewed. Looks good. Recommendation mapping to _deploying domain controllers_ - Would it be better if we specify in recommendation to deploy two domain controllers?

**Comment by oZakari** [comment]
> Reviewed. Looks good. Recommendation mapping to _deploying domain controllers_ - Would it be better if we specify in recommendation to deploy two domain controllers?

Thanks Swathi, I have adjusted the domain controller recommendation as follows for AVD:

image

**Comment by swathibhat1** [comment]
[like]  Swathi Bhat reacted to your message:
________________________________

**Pull request merged**